arch/arm/soc/nordic_nrf5/nrf52/power.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <soc_power.h>
+#include <nrf_power.h>
+
+/* System_OFF is deepest Power state available, On exiting from this
+ * state CPU including all peripherals reset
+ */
+static void _system_off(void)
+{
+	nrf_power_system_off();
+}
+
+void configure_ram_retention(void)
+{
+	/* Place Holder for RAM retention */
+}
+
+void _issue_low_power_command(void)
+{
+	__SEV();
+	__WFE();
+	__WFE();
+	if (NRF_GPIOTE->EVENTS_PORT) {
+		NRF_GPIOTE->EVENTS_PORT = 0;
+	}
+}
+
+static void _low_power_mode(enum power_states state)
+{
+	switch (state) {
+	case SYS_POWER_STATE_CPU_LPS:
+		nrf_power_task_trigger(NRF_POWER_TASK_CONSTLAT);
+		break;
+
+	case SYS_POWER_STATE_CPU_LPS_1:
+		nrf_power_task_trigger(NRF_POWER_TASK_LOWPWR);
+		break;
+
+	default:
+		/* Unsupported State */
+		break;
+	}
+
+	/* Wait for __WFE*/
+	_issue_low_power_command();
+
+}
+
+void _sys_soc_set_power_state(enum power_states state)
+{
+	configure_ram_retention();
+	switch (state) {
+	case SYS_POWER_STATE_CPU_LPS:
+		_low_power_mode(SYS_POWER_STATE_CPU_LPS);
+		break;
+	case SYS_POWER_STATE_CPU_LPS_1:
+		_low_power_mode(SYS_POWER_STATE_CPU_LPS_1);
+		break;
+#if (defined(CONFIG_SYS_POWER_DEEP_SLEEP))
+	case SYS_POWER_STATE_DEEP_SLEEP:
+		_system_off();
+		break;
+#endif
+	default:
+		/* Unsupported State */
+		break;
+	}
+}
+
+void _sys_soc_power_state_post_ops(enum power_states state)
+{
+	switch (state) {
+	case SYS_POWER_STATE_CPU_LPS:
+		break;
+	case SYS_POWER_STATE_CPU_LPS_1:
+		break;
+#if (defined(CONFIG_SYS_POWER_DEEP_SLEEP))
+	case SYS_POWER_STATE_DEEP_SLEEP_2:
+		break;
+	case SYS_POWER_STATE_MAX:
+		break;
+	case SYS_POWER_STATE_DEEP_SLEEP_1:
```

Comment:
_DEEP_SLEEP_1, _DEEP_SLEEP_2, and _MAX shouldn't be listed here, since by definition they won't need any post operations (first two are unsupported, the last one is a "do nothing").
But is there a point in providing all these cases when each one is empty? Wouldn't a comment like "Currently, there is no need to perform any specific operations after exiting low power states." be enough instead?
Currently, there iis no need to do any post ops